### ISSUE TITLE
Wrongly giving access with any authentication header

### ISSUE THREAD

**Issue by avadhootha** [sscarduzio/elasticsearch-readonlyrest-plugin#17]
Thanks for the plugin. I am facing an issue with basic authentication 
I wrote the access control rule to have basic authentication  in elasticseach configuration. 
Basic Authorisation key (clients should send this key base64-encoded in the 'Auth' header)
    > - name: full access if Basic HTTP auth
    >  type: allow
     >  auth_key: random_password
**auth_key** is in plain text

Now when I try to access the elasticsearch url without authentication it gives a null pointer exception ,that is good.

But I could access my elasticsearch with any random basic authentication header (even if doesn't match auth_key)  when i hit the url. 
Please help to resolve soon

**Comment by sscarduzio** [comment]
hold on I'll have a look

**Comment by sscarduzio** [comment]
Could you provide the full conf and the stack trace? I'm not in my machine now.

**Comment by sscarduzio** [comment]
No worries, I found the bug. Rebuilding the fix asap.

**Comment by sscarduzio** [comment]
@avadhootha the bug is fixed, so please:
* bin/plugin uninstall readonlyrest
* bin/plugin install https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/raw/master/download/elasticsearch-readonlyrest-v1.5_es-v2.1.1.zip

And test again. Let me know if you have any more issues.

**Comment by avadhootha** [comment]
Sorry for not coming here soon. I downloaded the rest plugin(version 1.5) now and tried to test it on elasticsearch 2.1.1.  

Now I don't seem to know how my elasticsearch has to be configured. I have tried giving the key both as base64 enocoded in elasticsearch.yml and as a plain text, with quotes and without quotes. Tried to hit the url with basic authentication header (base64encoded key of the auth_key written in the yml file )from the Rest client and nginx too.  Doesn't seem to work. It throws request forbidden error

What is the right way to configure in yml file?
 Should the auth_key parameter in yml file be user:password or password or base64encoded of user:password/password?
